### DISCUSSION TITLE
Help with Mod Integration with CC:Twaked 1.12.2 & Returning Lua Tables

### DISCUSSION THREAD

**Discussion by FireDragon91245** [cc-tweaked/CC-Tweaked#1553] [Q&A] [answered]
Hey im working on a CC:T Integration for my mod **(1.12.2)** and i cant figure out how to return a complex objekt aka a lua table
if i return a `LuaTable`, or any `class` they both show up as `nil` in game.

It is clear to my that the folowing would not work so i convertet the TestObj to a eqivalent LuaTable object but still comes up as nil.

```java
private static class TestObj{
  public int a = 12;
  public String b = "test";
  public boolean c = true;
}

class Test implements IPeripheral
{

...

  @Nullable
  @Override
  public Object[] callMethod(@Nonnull IComputerAccess iComputerAccess, @Nonnull ILuaContext iLuaContext, int i, @Nonnull Object[] objects)
  throws LuaException, InterruptedException {
    return new Object[]{ new TestObj(), new NulaTable() }; // both show up as nil in game
  }

...

}
```

Thanks in advance for any help :)

**Comment by SirEdvin** [comment, marked as answer]
You can just return `HashMap<String, Object>` and it will be transformed to Lua table

**Reply by SirEdvin** [reply]
For tuple of objects you can use `MethodResult` I guess

**Reply by FireDragon91245** [reply]
thx 💯 <3

**Comment by FireDragon91245** [comment]
Arrays and Arrays of HashMap<String, Object> also work
